plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/server/build/N4JSClusteringStorageAwareResourceLoader.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2022 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.server.build;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.n4js.n4JS.ImportDeclaration;
+import org.eclipse.n4js.n4JS.Script;
+import org.eclipse.n4js.utils.URIUtils;
+import org.eclipse.n4js.xtext.ide.server.build.XBuildContext;
+import org.eclipse.n4js.xtext.ide.server.build.XClusteringStorageAwareResourceLoader;
+import org.eclipse.n4js.xtext.workspace.ProjectConfigSnapshot;
+import org.eclipse.n4js.xtext.workspace.SourceFolderSnapshot;
+import org.eclipse.n4js.xtext.workspace.WorkspaceConfigAdapter;
+import org.eclipse.n4js.xtext.workspace.WorkspaceConfigSnapshot;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+/**
+ *
+ */
+public class N4JSClusteringStorageAwareResourceLoader extends XClusteringStorageAwareResourceLoader {
+
+	@Override
+	protected List<LoadResult> sort(XBuildContext context, List<LoadResult> results) {
```

Comment:
This would be a good place to provide a brief motivation for the sorting (i.e. avoid unnecessary chains of resolution / post-processing to limit memory footprint).

Reply:
Yes, thanks!